src/v/cluster/members_backend.cc

```diff
@@ -346,11 +382,28 @@ void members_backend::calculate_reallocations_after_node_added(
                   && to_move.left_to_move > 0) {
                     partition_reallocation reallocation(
                       model::ntp(tp_ns.ns, tp_ns.tp, p.id), p.replicas.size());
-
                     reallocation.replicas_to_remove.emplace(to_move.id);
+                    auto current_assignment = p;
+                    reassign_replicas(current_assignment, reallocation);
+                    // if this reassignment does not involve the node we are
+                    // targetting do not add it
+                    if (!is_reassigned_to_node(reallocation, meta.update.id)) {
+                        continue;
+                    }
```

Comment:
I'm curious: I'm not sure what our scheduling algorithm is, but as we fill the node, does it become more and more likely to hit this code path? I wonder if it makes sense to update `reassign_replicas()` to target movement towards a specific node.

Reply:
yes, this is very good point, but the allocator logic is taught to chose the least allocated node (soft constraint) so by the time the node is populated with partitions (the stop point) this should happen rarely